collector/collector.go

```diff
@@ -167,21 +164,43 @@ func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
 func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	e.mu.Lock() // ensure no simultaneous scrapes
 	defer e.mu.Unlock()
-	if e.config.ScrapeInterval == 0 { // if we are to scrape when the request is made
-		e.scrape(ch)
-	} else {
-		scrapeResults := e.scrapeResults // There is a risk that e.scrapeResults will be replaced while we traverse this look. This should mitigate that risk
-		for idx := range scrapeResults {
-			ch <- scrapeResults[idx]
-		}
-	}
+	e.scrape(ch)
```

Comment:
This makes it so that a scrape will always happen when the metrics endpoints is hit, even if ScrapeInterval is > 0.

Should this still be kept intact? 

I do think there should be a mux protecting e.scrapeResults, since there could be concurrent read/writes.

Reply:
True let me take a second pass at this